File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        unordered_map<int,int>one,two;
        int n=A.size();
        vector<int>ans(n,0);
        for(int i=0;i<n;i++)
        {
            ans[i]=i>=1?ans[i-1]:0;
            if(A[i]==B[i])
            {
                ans[i]++;
            }else
            {
                if(one[B[i]])
                    ans[i]++;
                if(two[A[i]])
                    ans[i]++;
            }
            one[A[i]]++;
            two[B[i]]++;
        }
        return ans;
    }
};

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {

        Set<Integer> a1 = new HashSet<>();
        Set<Integer> b1 = new HashSet<>();
        int common =0;
          int n = A.length;
        int[] ans = new int[n];

      

        for(int i=0;i<n;i++){

            a1.add(A[i]);
            b1.add(B[i]);

           
            if(a1.contains(B[i])) common++;
            if(b1.contains(A[i]) && A[i]!=B[i]) common++;

            ans[i] = common;


        }

        return ans;


        
    }
}